File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
while(True):
    ls_n = list(map(int,input().split()))
    ls_n.sort()

    if sum(ls_n) == 0:
        break

    if ls_n[2] >= ls_n[0] + ls_n[1]:
        print("Invalid")
    elif ls_n[0] == ls_n[1] and ls_n[1] == ls_n[2] and ls_n[2] == ls_n[0]:
        print("Equilateral")
    elif ls_n[0] == ls_n[1] or ls_n[1] == ls_n[2] or ls_n[2] == ls_n[0]:
        print("Isosceles")
    elif ls_n[0] != ls_n[1] and ls_n[1] != ls_n[2] and ls_n[2] != ls_n[0]:
        print("Scalene")

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

#define TRUE 1 
#define FALSE 0

int main() {
    int a, b, c;

    while(TRUE){
        scanf("%d %d %d", &a, &b, &c);

        if (a == 0 && b == 0 && c == 0) {
            break;
        }

        if (a > b) { int temp = a; a = b; b = temp; }
        if (b > c) { int temp = b; b = c; c = temp; }
        if (a > b) { int temp = a; a = b; b = temp; }

        if (a + b <= c) {
            printf("Invalid\n");
        }
        else if (a == b && b == c) {
            printf("Equilateral\n");
        }
        else if (a == b || b == c || a == c) {
            printf("Isosceles\n");
        }
        else {
            printf("Scalene\n");
        }
    }


    return 0;
}